internal/cli/base_init.go

```diff
@@ -63,6 +68,44 @@ func (c *baseCommand) initConfigLoad(path string) (*configpkg.Config, configpkg.
 	return cfg, results, nil
 }
 
+// Given a duration, check the clientContext's auth token to determine if the
+// token expires within the duration.
+// Emits a warning on the cli if token expires within the duration.
+// Returns any errors which may have occurred parsing the token.
+//
+// This function provides no guarantee that the token itself is valid as
+// validation checks can only happen on the server.
+func (c *baseCommand) checkTokenExpiry(duration time.Duration) error {
+	token := c.clientContext.Server.AuthToken
+	tokenMagic := "wp24"
```

Comment:
This might be stored as a const, maybe we can use this instead? I think it's this:

https://github.com/hashicorp/waypoint/blob/4ff589ab89363e83fbdf031e13382b7b42861268/pkg/tokenutil/decode.go#L17-L22

Actually maybe we could just use that whole function instead and use the return values to handle the `body.ValidUntil`?

https://github.com/hashicorp/waypoint/blob/4ff589ab89363e83fbdf031e13382b7b42861268/pkg/tokenutil/decode.go#L24-L27